Rules/UseConsistentCasing.cs

```diff
@@ -0,0 +1,235 @@
+        [ConfigurableRuleProperty(defaultValue: "lowercase")]
+        public string Case
+        {
+            get
+            {
+                return casing.ToString();
+            }
+            set
+            {
+                if (String.IsNullOrWhiteSpace(value) || !Enum.TryParse<Casing>(value, true, out casing))
+                {
+                    casing = Casing.lowercase;
+                }
+            }
+        }
+        */
+
+        /// <summary>
+        /// AnalyzeScript: Analyze the script to verify consistency of keyword and operator case
+        /// </summary>
+        public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
+            Token[] tokens;
+            ParseError[] errors;
+
+            if (CheckKeyword) {
+                // sorry, but we had to re-parse, because the AST doesn't give us keywords...
+                Parser.ParseInput(ast.Extent.Text, out tokens, out errors);
+
+                // Iterates all keywords and check the case
+                foreach (Token keyword in tokens.Where(t => t.TokenFlags.HasFlag(TokenFlags.Keyword)))
+                {
+                    string correctCase = keyword.Text.ToLowerInvariant();
+                    if (!keyword.Text.Equals(correctCase, StringComparison.Ordinal))
+                    {
+                        yield return GetDiagnosticRecord(keyword, fileName, correctCase);
+                    }
+                }
+            }
+
+            if (CheckOperator) {
+                // find them all at once for performance reasons
+                foreach (ExpressionAst oper in ast.FindAll(a => a is UnaryExpressionAst || a is BinaryExpressionAst, true))
+                {
+                    // but handle them separately
+                    UnaryExpressionAst unary = oper as UnaryExpressionAst;
+                    if (unary != null) {
+                        string correctCase = "-" + unary.TokenKind.ToString().ToLowerInvariant();
+                        string actualCase = unary.Extent.Text.Substring(0, correctCase.Length);
+                        if (!actualCase.Equals(correctCase, StringComparison.Ordinal))
+                        {
+                            yield return GetDiagnosticRecord(unary, fileName, correctCase);
+                        }
+                    }
+                    else
+                    {
+                        BinaryExpressionAst binary = oper as BinaryExpressionAst;
+                        string correctCase = binary.ErrorPosition.Text.ToLowerInvariant();
+                        if (!binary.ErrorPosition.Text.Equals(correctCase, StringComparison.Ordinal))
+                        {
+                            yield return GetDiagnosticRecord(binary, fileName, correctCase);
+                        }
+                    }
```

Comment:
```suggestion
                    switch (oper)
                    {
                        case UnaryExpressionAst unary:
                            string correctCase = "-" + unary.TokenKind.ToString().ToLowerInvariant();
                            string actualCase = unary.Extent.Text.Substring(0, correctCase.Length);
                            if (!actualCase.Equals(correctCase, StringComparison.Ordinal))
                            {
                                yield return GetDiagnosticRecord(unary, fileName, correctCase);
                            }
                            continue;

                        case BinaryExpressionAst binary:
                            string correctCase = binary.ErrorPosition.Text.ToLowerInvariant();
                            if (!binary.ErrorPosition.Text.Equals(correctCase, StringComparison.Ordinal))
                            {
                                yield return GetDiagnosticRecord(binary, fileName, correctCase);
                            }
                            continue;
                    }
```

Reply:
Also I don't understand the `binary.ErrorPosition.Text` code -- what's happening there? I think we should either add a comment explaining the trick or change the code to be like the unary operator version.

Reply:
I don't know why it's called "ErrorPosition" but in a BinaryExpressionAst, that's the ScriptExtent for the actual operator, so we're just .ToLower'ing it...

![image](https://user-images.githubusercontent.com/192942/129837096-d44cfefb-8c45-4fec-9546-baf4c91e6093.png)

It's weird, but it's **way** better than the UnaryExpressionAst, which doesn't _have_ an extent for the actual operator at all...

Originally I had done these using the `keywords` instead of `AST`, and now I've realized switching has introduced a bug and I may need to switch them back...